frontend/src/pages/warehouse/ModalEditWarehouse.js

```diff
@@ -0,0 +1,367 @@
+            .then(resp => resp.json())
+            .then(res => {
+                const optionsFromBack = res.map((dispatcher) => ({
+                    id: dispatcher.id,
+                    name: dispatcher.name,
+                    surname: dispatcher.surname,
+                    username: dispatcher.username
+                }));
+                setDispatcherOptions(optionsFromBack);
+            });
+    };
+
+    const handleSearchAllDispatchersById = (id) => {
+        fetch(`/customers/${props.currentCustomerId}/users/dispatchers/${id}?id=${id}`)
+            .then(resp => resp.json())
+            .then(res => {
+                const dispatchersFromBack = res.map((dispatcher) => ({
+                    id: dispatcher.id,
+                    name: dispatcher.name,
+                    surname: dispatcher.surname,
+                    username: dispatcher.username
+                }));
+                setDispatchers(dispatchersFromBack);
+            });
+    };
+
+    const onChangeState = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {
+                ...preState.addressDto,
+                state: (e.length ?
+                    {id: e[0].id, state: e[0].state}
+                    : {id: '', state: ''})
+            }
+        }));
+    };
+
+    const addDispatcher = (e) => {
+        if (e.length) {
+            const isContains = dispatchers.find(disp => e[0].id === disp.id)
+            if (!isContains) {
+                e.map(dispatcher =>
+                    setDispatchers(preState => ([
+                        ...preState, dispatcher
+                    ])));
+            }
+        }
+    };
+
+    const handleCity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, city: e.target.value}
+        }));
+    };
+
+    const handleLineOne = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine1: e.target.value}
+        }));
+    };
+
+    const handleLineTwo = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine2: e.target.value}
+        }));
+    };
+
+    const handleTotalCapacity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            totalCapacity: e.target.value
+        }));
+    };
+
+    const editWarehouseHandler = (e) => {
+        e.preventDefault();
+        const dispatchersId = dispatchers.map(dispatcher => dispatcher.id);
+        const updateWarehouseDto = {...warehouseDto, dispatchersId: dispatchersId, customerId: props.currentCustomerId}
+        deleteWarehouseFromDispatchers(dispatcherDeleteList);
+
+        let validationResult = validateEditWarehouse(updateWarehouseDto, dispatchersId);
```

Comment:
this line will be executed regardless of the deleteWarehouseFromDispatchers result. Is it right?

Reply:
yes. this method is executed independently

Reply:
the issue is not with this particular method. All the rest of the function is executed regardless of the result of delete. Don't we have cases where this is not appropriate?